js_modules/dagster-ui/packages/ui-core/src/runs/fetchPaginatedBucketData.ts

```diff
@@ -67,13 +67,18 @@ export async function fetchPaginatedData<DataType, CursorType, ErrorType>({
   let currentCursor: CursorType | undefined = undefined;
 
   while (hasMoreData) {
-    const {cursor, hasMore, data, error} = await fetchData(currentCursor);
-    if (error) {
-      throw error;
+    try {
+      const {cursor, hasMore, data, error} = await fetchData(currentCursor);
+      if (error) {
+        throw error;
+      }
+      dataSoFar.push(...data);
+      currentCursor = cursor;
+      hasMoreData = hasMore;
+    } catch (e) {
+      throw e;
+      hasMoreData = false;
```

Comment:
confused by set after throw, should it be the other way? what prompted this change?

Reply:
oops I'll remove this. This was when I was debugging 😅